pages/index.js

```diff
@@ -19,9 +21,20 @@ export default function Home() {
   function handleProgress(newProgress) {
     setProgress(newProgress);
   }
-
+  function handleAddValue() {
```

Comment:
Redundant code: you can reduce all your calculation functions to one single function by using parameters!